VoodooPS2Controller/VoodooPS2Controller.cpp

```diff
@@ -798,40 +844,25 @@ void ApplePS2Controller::uninstallInterruptAction(PS2DeviceType deviceType)
   // Is it the keyboard or the mouse interrupt handler that was requested?
   // We only uninstall it if it is currently installed.
 
-  if (deviceType == kDT_Keyboard && _interruptInstalledKeyboard)
+  if (port == kPS2KbdIdx && _interruptInstalledKeyboard)
   {
     setCommandByte(0, kCB_EnableKeyboardIRQ);
-#ifdef NEWIRQ
-    getProvider()->disableInterrupt(0);
-    getProvider()->unregisterInterrupt(0);
-#else
     getProvider()->disableInterrupt(kIRQ_Keyboard);
     getProvider()->unregisterInterrupt(kIRQ_Keyboard);
-#endif
-    _workLoop->removeEventSource(_interruptSourceKeyboard);
     _interruptInstalledKeyboard = false;
-    _interruptActionKeyboard = NULL;
-    _packetActionKeyboard = NULL;
-    _interruptTargetKeyboard->release();
-    _interruptTargetKeyboard = 0;
   }
 
-  else if (deviceType == kDT_Mouse && _interruptInstalledMouse)
+  else if (port > kPS2KbdIdx)
   {
-    setCommandByte(0, kCB_EnableMouseIRQ);
-#ifdef NEWIRQ
-    getProvider()->disableInterrupt(1);
-    getProvider()->unregisterInterrupt(1);
-#else
-    getProvider()->disableInterrupt(kIRQ_Mouse);
-    getProvider()->unregisterInterrupt(kIRQ_Mouse);
-#endif
-    _workLoop->removeEventSource(_interruptSourceMouse);
-    _interruptInstalledMouse = false;
-    _interruptActionMouse = NULL;
-    _packetActionMouse = NULL;
-    _interruptTargetMouse->release();
-    _interruptTargetMouse = 0;
+    _interruptInstalledMouse--;
```

Comment:
Put an assert here that it is not already 0. The previous code does check this, if you believe it is impossible, please at least add a debug check.